### ISSUE TITLE
cpost double events

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarrac#94]
It looks like cpost produces two posts for a single file creation under some circumstances.  It would account for a handful of files out of a thousand in the static_flow test.

These patches to the flow tests filter out that issue:

https://github.com/MetPX/sr_insects/commit/cba92a3467a6f3d3869b1e70bfc22e0d1283a05d

https://github.com/MetPX/sr_insects/commit/2aecd54c0a6177b5cc60dd93ef3ebe58e372ec30

revert the adjustments to the totcveille collation when this problem is addressed.

**Comment by petersilva** [comment]
as of now 2022/12/23, the CI/CD shim tests succeed on ubuntu 22.04, but fail one test on ubuntu 20.04 and 18.04.
Reproduced in a VM, it looks like this::

```

ubuntu@legendary-sylph:~/metpx-sr3c$ make test_shim |& grep RESULT
RESULT: Good! 1 sha512 posts from capturing stdout. as expected
RESULT: Good! 1 sha512 posts from c program run.. as expected
RESULT: Good! 1 sha512 posts from python program run. as expected
RESULT: Good! 1 sha512 posts from cp command. as expected
RESULT: BAD! 2 sha512 posts, expected: {'sha512': 1} for touch command
RESULT: Good! 1 link posts from symlink command. as expected
RESULT: Good! 1 rename posts from moving a symlink. as expected
RESULT: Good! 1 rename posts from hardlink to a symlink. as expected
RESULT: Good! 1 rename posts from moving a file.. as expected
RESULT: Good! 1 remove posts from removing a file.. as expected
RESULT: Good! 1 remove posts from removing a file.. as expected
RESULT: Good! 1 sha512 posts from stdout redirection in a subdir. as expected
RESULT: Good! 1 sha512 posts from stdout redirection in a subsubdir. as expected
RESULT: Good! 2 rename posts from renaming subdirs should cause file rename events.. as expected
RESULT: Good! 2 remove posts from removing a whole tree events.. as expected
RESULT: Good! 2 remove posts from removing two files. as expected
RESULT: summary: good: 15, bad 1, total 16
ubuntu@legendary-sylph:~/metpx-sr3c$

```

So the touch command is still seeing double posts, while everything else is good.

**Comment by petersilva** [comment]
duped_fd management properly fixed by: cc8817a223bbc29b0404c11925bbd97e9b25088c
